vmm/src/lib.rs

```diff
@@ -1488,114 +1511,97 @@ impl Vmm {
     }
 
     fn is_instance_initialized(&self) -> bool {
-        let instance_state = {
-            // Use expect() to crash if the other thread poisoned this lock.
-            let shared_info = self.shared_info.read()
-                .expect("Failed to determine if instance is initialized because shared info couldn't be read due to poisoned lock");
-            shared_info.state.clone()
-        };
-        match instance_state {
-            InstanceState::Uninitialized => false,
-            _ => true,
-        }
+        let error_string = "Failed to determine if instance is initialized because shared info couldn't be read due to poisoned lock";
+        self.shared_info.read().expect(error_string).state.clone() != InstanceState::Uninitialized
```

Comment:
I don't think you need clone here anymore.

Reply:
Indeed, I've removed it and the code seems to still work